horizon/data_manager/policy_store.py

```diff
@@ -0,0 +1,139 @@
+                            await self._insert_fact(
+                                "role_assignments",
+                                {
+                                    "actor": user_key,
+                                    "tenant": subject_key,
+                                    "role": role_key,
+                                    "resource": "",
+                                },
+                            )
+                        else:
+                            # TODO missing resource_type
+                            await self._insert_fact(
+                                "role_assignments",
+                                {
+                                    "actor": user_key,
+                                    "tenant": "",
+                                    "role": role_key,
+                                    "resource": subject_key,
+                                },
+                            )
+            case ["users", user_key]:
+                attributes = policy_data.get("attributes", {})
+                attributes.pop("key", None)
+                return await self._insert_fact(
+                    "users",
+                    {
+                        "key": user_key,
+                        "first_name": attributes.pop("first_name", ""),
+                        "last_name": attributes.pop("last_name", ""),
+                        "email": attributes.pop("email", ""),
+                        "attributes": attributes,
+                    },
+                )
+            case ["resource_instances", instance_key]:
+                # TODO missing resource_type
+                return await self._insert_fact(
+                    "resource_instance",
+                    {
+                        "key": instance_key,
+                        "attributes": policy_data.get("attributes", {}),
+                    },
+                )
+            case _:
+                return await super().set_policy_data(
+                    policy_data=policy_data, path=path, transaction_id=transaction_id
+                )
+
+    async def _insert_fact(self, fact_type: str, attributes: dict[str, str]):
+        try:
+            res = await self._client.post(
+                "/facts/insert",
+                json={
+                    "type": fact_type,
+                    "attributes": attributes,
+                },
+            )
+            if res.status != 200:
+                error = await res.text()
+                logger.error(f"Failed to insert fact: {res.status}\n{error}")
+            return res
+        except Exception as e:
+            logger.exception(f"Failed to insert fact: {e}")
+
+    async def delete_policy_data(
+        self, path: str = "", transaction_id: Optional[str] = None
+    ):
+        # TODO forward relevant objects to data manager instead of OPA
+        return super().delete_policy_data(path=path, transaction_id=transaction_id)
```

Comment:
TODO: this requires implementation of **deletes** into the factstore.
it's ok (even recommended) to merge this now (to speed up testing) but @danyi1212 please open a ticket so we remember to close this gap later this week.